many_to_one.go

```diff
@@ -62,27 +62,22 @@ func (d *ManyToOne) Set(data GenericDataType) {
 // TryNext will attempt to read from the next slot of the ring buffer.
 // If there is not data available, it will return (nil, false).
 func (d *ManyToOne) TryNext() (data GenericDataType, ok bool) {
-	readIndex := atomic.LoadUint64(&d.readIndex)
-	idx := readIndex % uint64(len(d.buffer))
-
-	value, ok := d.tryNext(idx)
-	if ok {
-		atomic.AddUint64(&d.readIndex, 1)
-	}
-	return value, ok
-}
-
-func (d *ManyToOne) tryNext(idx uint64) (GenericDataType, bool) {
+	idx := d.readIndex % uint64(len(d.buffer))
 	result := (*bucket)(atomic.SwapPointer(&d.buffer[idx], nil))
 
 	if result == nil {
 		return nil, false
 	}
+	if result.seq < d.readIndex {
+		return nil, false
+	}
 
 	if result.seq > d.readIndex {
-		d.alerter.Alert(int(result.seq - d.readIndex))
-		atomic.StoreUint64(&d.readIndex, result.seq)
+		dropped := result.seq - d.readIndex
+		d.readIndex = result.seq
+		d.alerter.Alert(int(dropped))
 	}
 
+	d.readIndex++
```

Comment:
This does not have to be atomic because it is one reader -- only one go routine is reading at a time.